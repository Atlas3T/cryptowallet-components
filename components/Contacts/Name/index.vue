<template>
  <div class="center-me">
    {{ contactData.displayName }}
  </div>
</template>

<script>
import { mapState } from 'vuex';

export default {
  name: 'Name',
  computed: {
    ...mapState({
      contacts: (state) => { return state.contacts.contacts; },
      id: (state) => { return state.route.params.id; },
    }),
    contactData() {
      return this.contacts.find((contact) => { return String(contact.id) === this.id; });
    },
  },
};
</script>

<style scoped>
  .center-me {
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 2rem;
  }
</style>
