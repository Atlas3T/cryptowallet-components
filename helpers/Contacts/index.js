import Vue from 'vue';

/**
 * Handles the contacts import and saves them in the store
 */
export default function Contacts() {
  this.vm = new Vue();
  this.import = (contacts) => {
    this.contacts = contacts;
    this.vm.$once('contactsImported', () => { return this.success(); });
    this.vm.$once('contactsImportFailure', () => { return this.failure(); });
    this.vm.$once('contactsImportError', (event) => { return this.error(event.payload); });

    try {
      const options = new ContactFindOptions();
      options.multiple = true;
      const fields = ['*'];
      navigator.contacts.find(fields, (contactsArray) => {
        this.contacts.contacts = contactsArray;
        this.vm.$emit('contactsImported');
      }, () => {
        this.vm.$emit('contactsImportFailure');
      }, options);
    } catch (err) {
      this.vm.$emit('contactsImportError', { payload: err });
    }

    return this;
  };

  this.on = (event, callback) => {
    this[event] = callback;
    return this;
  };
}
