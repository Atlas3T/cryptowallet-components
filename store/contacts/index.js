export default {
  namespaced: true,
  state: {
    contacts: [
      {
        id: 1,
        displayName: 'Terry Tibbs',
        address: 'ncousohfodicwjdclwdnc',
      },
    ],
  },
};
