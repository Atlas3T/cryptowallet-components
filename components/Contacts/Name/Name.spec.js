import { shallowMount } from '@vue/test-utils';
import Name from '@/components/Contacts/Name';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';

describe('Name.vue', () => {
  let storeMocks;
  let wrapper;
  let store;
  let router;

  function wrapperInit(options) {
    return shallowMount(Name, options);
  }

  function storeInit(custom) {
    storeMocks = createStoreMocks(custom);
    router = createRouter(storeMocks.store);
    router.push({ path: `/contact/1` });
    wrapper = wrapperInit({ router, localVue, store: storeMocks.store, i18n });
    store = wrapper.vm.$store;
  }

  beforeEach(() => storeInit());

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the contacts displayName based on the URL param { id }', () => {
    const displayName = store.state.contacts.contacts.find(contact => contact.id === 1).displayName;
    expect(wrapper.html().includes(displayName)).toBe(true);
  });
});
