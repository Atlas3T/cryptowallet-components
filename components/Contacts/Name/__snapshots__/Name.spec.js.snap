// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name.vue renders and matches snapshot 1`] = `
<div
  class="center-me"
>
  
  Terry Tibbs

</div>
`;
